Debian Bug#193634: mikmod: Won't play LHA-compressed MODs with spaces in their names

  - updated a couple of file list commands to produce better parseable output
  - changed MA_FindFiles to remove trailing spaces instead everyting beginning
    with the first space, to allow spaces in filenames in archives

diff -Ndur mikmod-3.1.6.orig/src/marchive.c mikmod-3.1.6/src/marchive.c
--- mikmod-3.1.6.orig/src/marchive.c	2003-06-03 14:32:02.000000000 +0200
+++ mikmod-3.1.6/src/marchive.c	2003-06-03 14:38:01.000000000 +0200
@@ -149,19 +149,19 @@
 /* PKzip archive */
 	{	0,	pksignature,	"unzip",	"-pqq", NULL,	"-vqq", NULL,	58},
 /* zoo */
-	{	20,	zoosignature,	"zoo",		"xpq", NULL,	"lq", NULL,		47},
+	{	20,	zoosignature,	"zoo",		"xpq", NULL,	"lfq", NULL,		0},
 /* rar */
 	{	0,	rarsignature,	"unrar",	"p", "-inul",	"v", "-c-",		1},
 /* lharc */
 	{	2,	lhsignature,	"lha",		"pq", NULL,		"vvq", NULL,	0},
 	{	2,	lzsignature,	"lha",		"pq", NULL,		"vvq", NULL,	0},
 /* tar */
-	{	257,tarsignature,	"tar",		"-xOf", NULL,	"-tRf", NULL,	18},
+	{	257,tarsignature,	"tar",		"-xOf", NULL,	"-tf", NULL,	0},
 /* tar.gz */
-	{	-1,	targzextension,	"tar",		"-xOzf", NULL,	"-tRzf", NULL,	18},
+	{	-1,	targzextension,	"tar",		"-xOzf", NULL,	"-tzf", NULL,	0},
 /* tar.bz2 */
 	{	-1,	tarbzip2extension,	"tar",	"--use-compress-program=bzip2",
-			"-xOf",		"--use-compress-program=bzip2",	"-tRf",			18},
+			"-xOf",		"--use-compress-program=bzip2",	"-tf",			0},
 /* gzip */
 	{	0,	gzsignature,	"gzip",		"-dqc", NULL,	NULL, NULL,		0},
 /* bzip2 */
@@ -568,6 +568,11 @@
 						while (fgets(string,PATH_MAX+MAXCOLUMN+1,file)) {
 							s=strlen(string)-1;
 							if (string[s]=='\n') string[s]='\0';
+							/* remove trailing whitespace from the filename */
+							for (s=0,t=0;string[t++];)
+								if (string[t]!=' ')
+									s=t;
+							string[s]='\0';
 							if (MA_isModuleFilename2(string+MA_archiver[archive].nameoffset))
 								PL_Add(pl,(string+MA_archiver[archive].nameoffset),filename,0,0);
 						}
